include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,811 @@
+                ( pluginPrefix + ".qx_max" ).c_str( ),
+                po::value< float_X >( & q_max[ 0 ] )->default_value( 5 ),
+                "reciprocal space range qx_max (A^-1)"
+            )(
+                ( pluginPrefix + ".qy_max" ).c_str( ),
+                po::value< float_X >( & q_max[ 1 ] )->default_value( 5 ),
+                "reciprocal space range qy_max (A^-1)"
+            )(
+                ( pluginPrefix + ".qx_min" ).c_str( ),
+                po::value< float_X >( & q_min[ 0 ] )->default_value( -5 ),
+                "reciprocal space range qx_min (A^-1)"
+            )(
+                ( pluginPrefix + ".qy_min" ).c_str( ),
+                po::value< float_X >( & q_min[ 1 ] )->default_value( -5 ),
+                "reciprocal space range qy_min (A^-1)"
+            )(
+                ( pluginPrefix + ".n_qx" ).c_str( ),
+                po::value< int >( & numVectors[ 0 ] )
+                    ->default_value( 100 ),
+                "number of qx"
+            )(
+                ( pluginPrefix + ".n_qy" ).c_str( ),
+                po::value< int >( & numVectors[ 1 ] )
+                    ->default_value( 100 ),
+                "number of qy"
+            )(
+                (pluginPrefix + ".file").c_str( ),
+                po::value< std::string >( & fileName )
+                    ->default_value( pluginName + "Output" ),
+                "output file name"
+            )(
+                (pluginPrefix + ".ext").c_str( ),
+                po::value< std::string >( & fileExtension )
+                    ->default_value( "bp" ),
+                "openPMD filename extension (this controls the backend "
+                "picked by the openPMD API)"
+            )(
+                (pluginPrefix + ".compression").c_str( ),
+                po::value< std::string >( & compressionMethod )
+                    ->default_value( "" ),
+                "Backend-specific openPMD compression method, e.g., zlib "
+                "(see `adios_config -m` for help)"
+            )
+            (
+                (pluginPrefix + ".memoryLayout").c_str( ),
+                po::value< std::string >( & memoryLayout )
+                    ->default_value( "mirror" ),
+                "Possible values: 'mirror' and 'split'"
+                "Output can be mirrored on all Host+Device pairs or"
+                " uniformly split over all nodes. Split can be used "
+                "when the output array is to big to store the complete "
+                "computed q-space on one device."
+            );
+        }
+
+
+        std::string pluginGetName( ) const override { return pluginName; }
+
+
+        void setMappingDescription( MappingDesc * cellDescription ) override
+        {
+            this->cellDescription = cellDescription;
+        }
+
+
+        void restart(
+            uint32_t timeStep,
+            const std::string restartDirectory
+        ) override
+        {
+            // Keep this empty for now.
```

Comment:
This comment is not clear: is restarting not supported currently, or nothing has to be done here for restarting? Same for checkpointing just below.

Reply:
The restarting is not supported for now. I took it from the previous PRs. I would suggest to just change the parent class to `ILightweightPlugin` and remove this override completely. Could it brake my plugin in any way or is the restart support the only difference between the lightweight and the normal plugins?